cli/main.go

```diff
@@ -202,3 +220,31 @@ func getEnvInt(key string, defaultValue int) int {
 	}
 	return defaultValue
 }
+
+func getEnvFloat64(key string, defaultValue float64) float64 {
+	if value, ok := os.LookupEnv(key); ok {
+		val, err := strconv.ParseFloat(value, 64)
+		if err == nil {
+			return val
+		}
+	}
+	return defaultValue
+}
+
+func floatEthTo256Wei(val float64) (*types.U256Str, error) {
```

Comment:
maybe add some tests for this function

Reply:
Added a test but there are some potential rounding errors in the function, I don't think they impact the functionality but Ill update this after they are fixed.